src/NServiceBus.Core/DelayedDelivery/DelayedDeliveryOptionExtensions.cs

```diff
@@ -20,12 +18,12 @@ public static void DelayDeliveryWith(this SendOptions options, TimeSpan delay)
             Guard.AgainstNull(nameof(options), options);
             Guard.AgainstNegative(nameof(delay), delay);
 
-            if (options.GetExtensions().TryGetDeliveryConstraint(out DoNotDeliverBefore _))
+            if (options.DelayedDeliveryConstraint is DoNotDeliverBefore)
```

Comment:
Now that we aren't just adding delivery constraints to the bag for each call, we could remove this check. That means the last one called would be the one that gets used.

Reply:
Had a chat with @bording about this comment. It makes sense to get rid of these `if`s.

Reply:
Though to be 100% completely safe, maybe we should keep them.

Reply:
Last one wins make sense to me. It could be argued that it would be a breaking change to remove but I can't see how a user could rely on us throwing so I'm 👍 to remove since that code will always be causing confusion and adding an issue to remove in v8 seems overkill?

Reply:
👎 for removing the check. I still think it makes sense to prevent setups like
```
options.DelayDeliveryWith(...)
options.DoNotDeliverBefore(...)
```

for calls to the same API, last-one-wins is already active.

Reply:
@timbussmann I also thought that we should guard against this scenario and throw early rather than users to discover they are doing something wrong much later in the game.

Assuming I'm sending messages and set delayed delivery constraint conditionally, I'd probably create two `SendOption`s and use accordingly. Will not call the two methods on the same object to alternate between delayed delivery constraints. What would be the use case where both options would be applied?

Reply:
> What would be the use case where both options would be applied?

none. That would very likely be a mistake (maybe by reusing some code) which we should provide.